src/main/java/com/squareup/javapoet/AnnotationSpec.java

```diff
@@ -183,13 +185,17 @@ public Builder toBuilder() {
   }
 
   @Override public boolean equals(Object o) {
-    return o instanceof AnnotationSpec
-        && ((AnnotationSpec) o).type.equals(type)
-        && ((AnnotationSpec) o).members.equals(members);
+    if (this == o)
+      return true;
+    if (o == null)
+      return false;
+    if (getClass() != o.getClass())
+      return false;
+    return hashCode == o.hashCode();
```

Comment:
Ja. Correct. Will convert that incorrect optimization to `String.equals()` calls.